lib/rubygems/core_ext/kernel_require.rb

```diff
@@ -49,9 +49,12 @@ def require path
     # If there are no unresolved deps, then we can use just try
     # normal require handle loading a gem from the rescue below.
 
-    if Gem::Specification.unresolved_deps.empty? then
+    begin
       RUBYGEMS_ACTIVATION_MONITOR.exit
-      return gem_original_require(path)
+      original = gem_original_require(path)
+      return original
+    rescue LoadError
```

Comment:
Isn't this going to cause many exceptions during `require`?  Seems like it would make `-d` very noisy.  Is there another way to fix this?